modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java

```diff
@@ -102,6 +104,17 @@ protected static void writeException(JsonGenerator jg, Throwable t, MediaType me
         jg.flush();
     }
 
+    /**
+     * @since 11.5
+     */
+    protected static String getExceptionMessage(String exceptionMessage, int statusCode) {
+        if (statusCode < SC_INTERNAL_SERVER_ERROR || jsonFactoryManager.isStackDisplay() || Framework.isDevModeSet()) {
```

Comment:
Maybe a stupid question: what about 502, 503, 504 errors? They wil have the same `"Internal Server Error"` message?

Reply:
Well, it is the desired behavior. :)

Reply:
So, no question? 😬

Reply:
Indeed we could try to keep better specific messages for >500 but the http code is what's important to clients. And there's very few places in Nuxeo where we return something else than 500.